File: CS/WebSite/App_Code/ExportHelper.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Web.ASPxGridView.Export;
using DevExpress.Web.ASPxGridView;
using System.Web;
using System.IO;
using DevExpress.XtraPrinting;

namespace Grid
{
    public static class ExportHelper
    {
		#region Methods (3) 

		// Public Methods (3) 

         public static void ExportData(ASPxGridViewExporter exporter, ExportType type)
        {
            if (exporter != null)
            {
                exporter.DataBind();
                switch (type)
                {
                    case ExportType.XLS:
                        exporter.WriteXlsToResponse();
                        break;
                    case ExportType.PDF:
                        exporter.WritePdfToResponse();
                        break;
                    case ExportType.CSV:
                        exporter.WriteCsvToResponse();
                        break;
                    case ExportType.RTF:
                        exporter.WriteRtfToResponse();
                        break;
                    default:
                        break;
                }
            }
        }

        public static byte[] GetExportData(ASPxGridViewExporter exporter, ExportType type)
        {
            byte[] data = null;
            if (exporter != null)
            {
                exporter.DataBind();
                using (MemoryStream ms = new MemoryStream())
                {
                    switch (type)
                    {
                        case ExportType.XLS:
                            exporter.WriteXls(ms);
                            break;
                        case ExportType.PDF:
                            exporter.WritePdf(ms);
                            break;
                        case ExportType.CSV:
                            exporter.WriteCsv(ms);
                            break;
                        case ExportType.RTF:
                            exporter.WriteRtf(ms);
                            break;
                        default:
                            break;
                    }
                    data = ms.GetBuffer();
                }
            }
            return data;
        }

        public static void WriteDataToResponse(byte[] data, ExportType type)
        {
            if (data != null && data.Length > 0 && type != ExportType.none)
            {
                String fileEnding = String.Empty;
                String fileContent = String.Empty;
                switch (type)
                {
                    case ExportType.XLS:
                        fileContent = "application/ms-excel";
                        fileEnding = "xls";
                        break;
                    case ExportType.PDF:
                        fileContent = "application/pdf";
                        fileEnding = "pdf";
                        break;
                    case ExportType.CSV:
                        fileContent = "text/plain";
                        fileEnding = "csv";
                        break;
                    case ExportType.RTF:
                        fileContent = "text/enriched";
                        fileEnding = "rtf";
                        break;
                }
                HttpContext.Current.Response.Clear();
                HttpContext.Current.Response.Buffer = false;
                HttpContext.Current.Response.ClearHeaders();
                HttpContext.Current.Response.AppendHeader("Content-Type", fileContent);
                HttpContext.Current.Response.AppendHeader("Content-Transfer-Encoding", "binary");
                HttpContext.Current.Response.AppendHeader("Content-Disposition", "attachment; filename=Export." + fileEnding);
                HttpContext.Current.Response.BinaryWrite(data);
                HttpContext.Current.Response.End();
            }
        }

		#endregion Methods 
    }
}

File: VB/WebSite/App_Code/ExportHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Web.ASPxGridView.Export
Imports DevExpress.Web.ASPxGridView
Imports System.Web
Imports System.IO
Imports DevExpress.XtraPrinting

Namespace Grid
	Public NotInheritable Class ExportHelper
        ' Public Methods (3) 

        Private Sub New()
        End Sub
        Public Shared Sub ExportData(ByVal exporter As ASPxGridViewExporter, ByVal type As ExportType)
            If exporter IsNot Nothing Then
                exporter.DataBind()
                Select Case type
                    Case ExportType.XLS
                        exporter.WriteXlsToResponse()
                    Case ExportType.PDF
                        exporter.WritePdfToResponse()
                    Case ExportType.CSV
                        exporter.WriteCsvToResponse()
                    Case ExportType.RTF
                        exporter.WriteRtfToResponse()
                    Case Else
                End Select
            End If
        End Sub

        Public Shared Function GetExportData(ByVal exporter As ASPxGridViewExporter, ByVal type As ExportType) As Byte()
            Dim data() As Byte = Nothing
            If exporter IsNot Nothing Then
                exporter.DataBind()
                Using ms As New MemoryStream()
                    Select Case type
                        Case ExportType.XLS
                            exporter.WriteXls(ms)
                        Case ExportType.PDF
                            exporter.WritePdf(ms)
                        Case ExportType.CSV
                            exporter.WriteCsv(ms)
                        Case ExportType.RTF
                            exporter.WriteRtf(ms)
                        Case Else
                    End Select
                    data = ms.GetBuffer()
                End Using
            End If
            Return data
        End Function

        Public Shared Sub WriteDataToResponse(ByVal data() As Byte, ByVal type As ExportType)
            If data IsNot Nothing AndAlso data.Length > 0 AndAlso type <> ExportType.none Then
                Dim fileEnding As String = String.Empty
                Dim fileContent As String = String.Empty
                Select Case type
                    Case ExportType.XLS
                        fileContent = "application/ms-excel"
                        fileEnding = "xls"
                    Case ExportType.PDF
                        fileContent = "application/pdf"
                        fileEnding = "pdf"
                    Case ExportType.CSV
                        fileContent = "text/plain"
                        fileEnding = "csv"
                    Case ExportType.RTF
                        fileContent = "text/enriched"
                        fileEnding = "rtf"
                End Select
                HttpContext.Current.Response.Clear()
                HttpContext.Current.Response.Buffer = False
                HttpContext.Current.Response.ClearHeaders()
                HttpContext.Current.Response.AppendHeader("Content-Type", fileContent)
                HttpContext.Current.Response.AppendHeader("Content-Transfer-Encoding", "binary")
                HttpContext.Current.Response.AppendHeader("Content-Disposition", "attachment; filename=Export." & fileEnding)
                HttpContext.Current.Response.BinaryWrite(data)
                HttpContext.Current.Response.End()
            End If
        End Sub

    End Class
End Namespace
